### ISSUE TITLE
webpack-dev-server outputPath error

### ISSUE THREAD

**Issue by zgabievi** [JeffreyWay/laravel-mix#150]
On command `npm run hot` I'm getting this error:
> ERROR in [copy-webpack-plugin] Using older versions of webpack-dev-server, devServer.outputPath must be defined to write to absolute paths

**Comment by JeffreyWay** [comment]
We've updated webpack-dev-server on master. It'll be tagged later today.

**Comment by zgabievi** [comment]
Thanks. 👍

**Comment by jbreuer95** [comment]
@zgabievi does it work for you now? I still get the same error

**Comment by zgabievi** [comment]
I'm not getting error now. It's working, but not for laravel project. In browser (http://localhost:8080), I'm getting this output `Cannot GET /`

**Comment by jbreuer95** [comment]
@zgabievi try using laravel valet.

**Comment by adriaanzon** [comment]
@zgabievi webpack-dev-server doesn't serve your Laravel app, only your assets. Use the `mix()` helper provided with Laravel. https://github.com/JeffreyWay/laravel-mix/blob/master/docs/hot-module-replacement.md#usage-in-laravel

**Comment by zgabievi** [comment]
@jbreuer95 I'am using Valet.

@adriaanzon Yes, I'm using **mix()** helper for assets

**Comment by adriaanzon** [comment]
It doesn't work when you visit your site the normal way? i.e. via `site.dev`, not via `localhost:8080`

**Comment by acacha** [comment]
Still having same problem with version:
```
$npm list webpack-dev-server
/home/sergi/Code/scool/curriculum_test
└─┬ laravel-mix@0.5.16
  └── webpack-dev-server@2.2.0 
```

**Comment by claytonrcarter** [comment]
I get this error if I have a `mix.copy()` command in my `webpack.mix.js`, like this:

    mix.copy('node_modules/vue/dist/vue.min.js', 'public/js' )

When I comment that out, it goes away.

The full error is:

```
chunk    {0} /js/app.js, /css/app.css (/js/app) 1.5 MB [entry] [rendered]
   [55] (webpack)/hot/emitter.js 77 bytes {0} [built]
   [56] ./resources/assets/js/app.js 1.52 kB {0} [built]
   [57] ./resources/assets/sass/app.scss 41 bytes {0} [built]
   [58] (webpack)-dev-server/client?http://localhost:8080 4.66 kB {0} [built]
   [59] (webpack)/hot/dev-server.js 1.57 kB {0} [built]
   [85] ./resources/assets/js/bootstrap.js 1.31 kB {0} [built]
   [86] ./resources/assets/js/vuex/store.js 13.7 kB {0} [built]
  [126] ./~/url/url.js 23.3 kB {0} [built]
  [128] ./resources/assets/js/components/CollationColumn.vue 1.65 kB {0} [built]
  [129] ./resources/assets/js/components/CollationInfo.vue 1.64 kB {0} [built]
  [130] ./resources/assets/js/components/CollationItemSearchModal.vue 1.69 kB {0} [built]
  [131] ./resources/assets/js/components/CollationLine.vue 1.64 kB {0} [built]
  [146] (webpack)-dev-server/client/socket.js 856 bytes {0} [built]
  [148] (webpack)/hot/log-apply-result.js 1.02 kB {0} [built]
  [149] multi (webpack)-dev-server/client?http://localhost:8080 webpack/hot/dev-server ./resources/assets/js/app.js ./resources/assets/sass/app.scss 64 bytes {0} [built]
     + 135 hidden modules

ERROR in [copy-webpack-plugin] Using older versions of webpack-dev-server, devServer.outputPath must be defined to write to absolute paths
Child extract-text-webpack-plugin:
    chunk    {0} extract-text-webpack-plugin-output-filename 698 kB [entry] [rendered]
        [0] ./~/bootstrap-sass/assets/fonts/bootstrap/glyphicons-halflings-regular.eot 92 bytes {0} [built]
        [1] ./~/css-loader/lib/css-base.js 1.51 kB {0} [built]
        [2] ./~/bootstrap-sass/assets/fonts/bootstrap/glyphicons-halflings-regular.svg 92 bytes {0} [built]
        [3] ./~/bootstrap-sass/assets/fonts/bootstrap/glyphicons-halflings-regular.ttf 92 bytes {0} [built]
        [4] ./~/bootstrap-sass/assets/fonts/bootstrap/glyphicons-halflings-regular.woff 93 bytes {0} [built]
        [5] ./~/bootstrap-sass/assets/fonts/bootstrap/glyphicons-halflings-regular.woff2 94 bytes {0} [built]
        [6] ./~/css-loader!./~/postcss-loader!./~/resolve-url-loader!./~/sass-loader?sourceMap&precision=8!./resources/assets/sass/app.scss 696 kB {0} [built]
```

**Comment by matrunchyk** [comment]
Same here

![Alt text](https://monosnap.com/file/MzSxCmWc4VrYH3dQKWmp07jb7ydgLb.png)

**Comment by alairock** [comment]
This is still happening on a fresh installation of Laravel Spark.

**Comment by JeffreyWay** [comment]
Make sure you have the latest version of Mix installed.

**Comment by alairock** [comment]
Indeed I have. 0.8.8.

**Comment by jerguslejko** [comment]
Same thing here

**Comment by setefocos** [comment]
Same here.
![error](https://cloud.githubusercontent.com/assets/26557942/24175394/b63d258a-0e74-11e7-8567-ee748ada2679.JPG)

**Comment by flexgrip** [comment]
Any progress on this or is there any fix? Fresh install of spark, this fails out of the box.

**Comment by joma74** [comment]
Same here. IMO https://github.com/kevlened/copy-webpack-plugin/issues/44 requires a now invalid option 
```bash
Invalid configuration object. webpack-dev-server has been initialised using a configuration object that does not
match the API schema.
 - configuration has an unknown property 'outputPath'. These properties are valid:
   object { hot?, hotOnly?, lazy?, host?, filename?, publicPath?, port?, socket?, watchOptions?, headers?, client
LogLevel?, overlay?, key?, cert?, ca?, pfx?, pfxPassphrase?, inline?, public?, https?, contentBase?, watchContent
Base?, open?, features?, compress?, proxy?, historyApiFallback?, staticOptions?, setup?, stats?, reporter?, noInf
o?, quiet?, serverSideRender?, index?, log?, warn? }
```
Breaks devserver reload badly, no workaround?!

**Comment by jbruni** [comment]
My temporary solution is to wrap the mix.copy entries in a condition, as follows:

if (mix.config.inProduction) {
    mix.copy('node_modules/bootstrap/fonts', 'public/fonts');
}

Of course, to have the assets copied, it is required to use `npm run production` once.

**Comment by JeffreyWay** [comment]
Fixed in 329ed0ccffb51d52111ec11359f2617c528c60c0

**Comment by matrunchyk** [comment]
Fantastic! Thanks!
